kura/org.eclipse.kura.rest.system.provider/src/main/java/org/eclipse/kura/rest/system/dto/FrameworkPropertiesDTO.java

```diff
@@ -0,0 +1,301 @@
+package org.eclipse.kura.rest.system.dto;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.eclipse.kura.system.SystemService;
+
+@SuppressWarnings("unused")
+public class FrameworkPropertiesDTO {
+
+    // hardware
+    private String biosVersion;
+    private String cpuVersion;
+    private String deviceName;
+    private String modelId;
+    private String modelName;
+    private String partNumber;
+    private String platform;
+    private Integer numberOfProcessors;
+    private Long totalMemory;
+    private Long freeMemory;
+    private String serialNumber;
+
+    // java
+    private String javaHome;
+    private String javaVendor;
+    private String javaVersion;
+    private String javaVmInfo;
+    private String javaVmName;
+    private String javaVmVersion;
+
+    // os
+    private String osArch;
+    private String osDistro;
+    private String osDistroVersion;
+    private String osName;
+    private String osVersion;
+    private Boolean isLegacyBluetoothBeaconScan;
+    private Boolean isLegacyPPPLoggingEnabled;
+    private String primaryMacAddress;
+    private String primaryNetworkInterfaceName;
+    private String fileSeparator;
+    private String firmwareVersion;
+
+    // kura
+    private String kuraDataDirectory;
+    private String kuraFrameworkConfigDirectory;
+    private String kuraHomeDirectory;
+    private String kuraMarketplaceCompatibilityVersion;
+    private Integer kuraSnapshotsCount;
+    private String kuraSnapshotsDirectory;
+    private String kuraStyleDirectory;
+    private String kuraTemporaryConfigDirectory;
+    private String kuraUserConfigDirectory;
+    private String kuraVersion;
+    private Boolean kuraHaveWebInterface;
+    private Boolean kuraHaveNetAdmin;
+    private Integer kuraWifiTopChannel;
+    private String kuraDefaultNetVirtualDevicesConfig;
+
+    // osgi
+    private String osgiFirmwareName;
+    private String osgiFirmwareVersion;
+
+    // command
+    private String commandUser;
+    private Integer commandZipMaxUploadNumber;
+    private Integer commandZipMaxUploadSize;
+
+    public FrameworkPropertiesDTO(SystemService systemService) {
+        populateProperties(systemService, "[\\w]+");
+    }
+
+    public FrameworkPropertiesDTO(SystemService systemService, List<String> names) {
+        for (String name : names) {
+            populateProperties(systemService, name);
+        }
+    }
+
+    private void populateProperties(SystemService systemService, String regex) {
+        populateHardwareProperties(systemService, regex);
+        populateJavaProperties(systemService, regex);
+        populateOsProperties(systemService, regex);
+        populateKuraProperties(systemService, regex);
+        populateOsgiProperties(systemService, regex);
+        populateCommandProperties(systemService, regex);
+    }
+
+    private void populateHardwareProperties(SystemService systemService, String regex) {
```

Comment:
Maybe you can use `Predicate<String> filter` instead of compiling and matching a regex multiple times, it should become a bit faster, in this way:

Check to add the property:
```
if (filter.test("biosVersion")) {
            this.biosVersion = systemService.getBiosVersion();
        }
```
Get all properties:
```
public FrameworkPropertiesDTO(SystemService systemService) {
        populateProperties(systemService, s -> true);
    }
```
Get filtered properties:
```
public FrameworkPropertiesDTO(SystemService systemService, List<String> names) {
            populateProperties(systemService, names::contains);
    }
```